conans/test/unittests/client/tools/scm/test_scm_base.py

```diff
@@ -53,4 +56,79 @@ def test_local_windows(self):
     def test_svn_ssh(self):
         self.assertEqual('svn+ssh://10.106.191.164/home/svn/shproject',
                          SCMBase._remove_credentials_url(
-                             'svn+ssh://username:password@10.106.191.164/home/svn/shproject'))
\ No newline at end of file
+                             'svn+ssh://username:password@10.106.191.164/home/svn/shproject'))
+
+
+class OutputMock(object):
+
+    def __init__(self):
+        self.out = list()
+
+    def warn(self, text):
+        self.out.append("WARN: " + text)
+
+
+class GetUrlWithCredentialsTest(unittest.TestCase):
+
+    def test_ssh(self):
+        scm = SCMBase()
+        self.assertEqual('ssh://github.com/conan-io/conan.git',
+                         scm.get_url_with_credentials("ssh://github.com/conan-io/conan.git"))
+
+    def test_ssh_username_password(self):
+        scm = SCMBase(username="dani", password="pass")
+        self.assertEqual('ssh://dani:pass@github.com/conan-io/conan.git',
+                         scm.get_url_with_credentials("ssh://github.com/conan-io/conan.git"))
+
+    def test_ssh_username(self):
+        scm = SCMBase(username="dani")
+        self.assertEqual('ssh://dani@github.com/conan-io/conan.git',
+                         scm.get_url_with_credentials("ssh://github.com/conan-io/conan.git"))
+
+    def test_ssh_password(self):
+        output = OutputMock()
+        scm = SCMBase(password="pass", output=output)
+        self.assertEqual('ssh://github.com/conan-io/conan.git',
+                         scm.get_url_with_credentials("ssh://github.com/conan-io/conan.git"))
+        self.assertIn("WARN: SCM username undefined, ignoring 'password' parameter", output.out)
+
+    def test_ssh_url_with_username_only_password(self):
+        scm = SCMBase(password="pass")
+        self.assertEqual('ssh://git:pass@github.com/conan-io/conan.git',
```

Comment:
As we talked, you can't add the password in ssh url's

Reply:
yes, I have been investigating a bit, and I haven't found any evidence this is supported.